Portal/src/Datahub.Portal/Pages/Home/HomePage.razor

```diff
@@ -87,5 +100,72 @@
             await _telemetryService.LogTelemetryEvent(TelemetryEvents.UserViewOtherProfile);
         }
     }
+    
+    protected override async Task OnAfterRenderAsync(bool firstRender)
+    {
+        await base.OnAfterRenderAsync(firstRender);
+
+        if (firstRender)
+        {
+            var language = await _userSettingsService.GetUserLanguage();
+            if (string.IsNullOrWhiteSpace(language))
+            {
+                await PromptForLanguage();
+            }
+
+            var userAcceptedTac = await _userSettingsService.HasUserAcceptedTAC();
+            if (!userAcceptedTac)
+            {
+                await PromptForTermsAndConditions();
+            }
+        }
+    }
+
+    private async Task PromptForTermsAndConditions()
+    {
+        var options = new DialogOptions
+        {
+            CloseOnEscapeKey = false,
+            DisableBackdropClick = true,
+        };
+
+        var tocDialog = await _dialogService.ShowAsync<UserTermsAndConditionsDialog>(Localizer["TermsAndConditions"], options);
+        var tocDialogResult = await tocDialog.Result;
+        if (!tocDialogResult.Canceled && tocDialogResult.Data is true)
+        {
+            await _userSettingsService.RegisterUserTAC();
+            await SetupRootFolder();
+            _isUserValid = true;
+        }
+        else
+        {
+            _navigationManager.NavigateTo(LogoutUrl, forceLoad: true);
+        }
+    }
+
+    private async Task PromptForLanguage()
+    {
+        var options = new DialogOptions
+        {
+            CloseOnEscapeKey = false,
+            DisableBackdropClick = true,
+        };
+
+        var languageDialog = await _dialogService.ShowAsync<UserSelectLanguageDialog>(Localizer["LanguagePreferencePrompt"], options);
+        var languageDialogResult = await languageDialog.Result;
+        if (!languageDialogResult.Canceled)
+        {
+            var selectedLanguage = languageDialogResult.Data.ToString();
+            var language = selectedLanguage?.Equals("english", StringComparison.InvariantCultureIgnoreCase) ?? true ? "en-CA" : "fr-CA";
+            await _userSettingsService.RegisterUserLanguage(language);
+            await _userSettingsService.SetLanguage(language);
+        }
+    }
```

Comment:
Nice, let's move this to MainLayout.razor in case they don't land on the home page first

Reply:
Solved with newest commit